### ISSUE TITLE
build failure on AIX 6.1

### ISSUE THREAD

**Issue by goshhhy** [ninja-build/ninja#1618]
found while trying to test issue #1513 and tested on master, and the issue turned out to be in master as well.

```
-bash-4.4$ uname -a
AIX aix 1 6 00F774544C00
```

```
-bash-4.4$ git clone https://github.com/ninja-build/ninja ninja-orig
Cloning into 'ninja-orig'...
remote: Enumerating objects: 10332, done.
Receiving objectsremote: Total 10332 (delta 0), reused 0 (delta 0), pack-reused 10332
Receiving objects: 100% (10332/10332), 3.67 MiB | 4.39 MiB/s, done.
Resolving deltas: 100% (7349/7349), done.
Checking connectivity... done.
-bash-4.4$ cd ninja-orig/
-bash-4.4$ ls
COPYING       HACKING.md    README        RELEASING     appveyor.yml  bootstrap.py  configure.py  doc           misc          src
-bash-4.4$ python ./configure.py --bootstrap
bootstrapping ninja...
warning: A compatible version of re2c (>= 0.11.3) was not found; changes to src/*.in.cc will not affect your build.
wrote build.ninja.
bootstrap complete.  rebuilding...
[28/28] LINK ninja
-bash-4.4$ ls
COPYING          README           appveyor.yml     build            configure.py     misc             ninja.bootstrap
HACKING.md       RELEASING        bootstrap.py     build.ninja      doc              ninja            src
-bash-4.4$ ninja ninja_test
[3/19] CXX build/test.o
FAILED: build/test.o 
g++ -MMD -MT build/test.o -MF build/test.o.d -g -Wall -Wextra -Wno-deprecated -Wno-missing-field-initializers -Wno-unused-parameter -fno-rtti -fno-exceptions -pipe '-DNINJA_PYTHON="python"' -O2 -DNDEBUG -fdiagnostics-color -D__STDC_FORMAT_MACROS -D_LARGE_FILES -I. -c src/test.cc -o build/test.o
src/test.cc: In member function 'void ScopedTempDir::CreateAndEnter(const string&)':
src/test.cc:208:41: error: 'mkdtemp' was not declared in this scope
   char* tempname = mkdtemp(name_template);
                                         ^
[18/19] CXX build/build_test.o
ninja: build stopped: subcommand failed.
-bash-4.4$ 
```

**Comment by NattyNarwhal** [comment]
AIX 6.1 does have `mkdtemp` (they added it in a patch), but the problem is that GCC's fixincludes "helpfully" removes things from the system header files.

```
$ grep -R mkdtemp /usr/include/stdlib.h                                                                                                                                                                                                   
        extern char *mkdtemp();
        extern char *mkdtemp(char *);
```

What I did was either:

1. Use GLib (likely not desirable here)

2. Declare the prototype yourself and the linker will handle it fine

**Comment by goshhhy** [comment]
@NattyNarwhal I tried this, and the linker does not in fact handle it fine. 

```
-bash-4.4$ ninja ninja_test
[2/2] LINK ninja_test
FAILED: ninja_test 
g++ -Lbuild -o ninja_test build/build_log_test.o build/build_test.o build/clean_test.o build/clparser_test.o build/depfile_parser_test.o build/deps_log_test.o build/dyndep_parser_test.o build/disk_interface_test.o build/edit_distance_test.o build/graph_test.o build/lexer_test.o build/manifest_parser_test.o build/ninja_test.o build/state_test.o build/string_piece_util_test.o build/subprocess_test.o build/test.o build/util_test.o -lninja -lperfstat
ld: 0711-317 ERROR: Undefined symbol: .mkdtemp(char*)
ld: 0711-345 Use the -bloadmap or -bnoquiet option to obtain more information.
collect2: error: ld returned 8 exit status
ninja: build stopped: subcommand failed.
-bash-4.4$ 
```

**Comment by NattyNarwhal** [comment]
It needs to be in an `extern "C"` block, otherwise it tries to resolve it to a C++ symbol instead...

**Comment by goshhhy** [comment]
that did get ninja_test to build, thanks - i'm strictly a C developer so i'm not used to having to deal with C++'s various idiosyncrasies

**Comment by jhasse** [comment]
Fixed by #1622.
